codewatch/helpers/inference.py

```diff
@@ -0,0 +1,123 @@
+from astroid import (
+    Load as LoadContext,
+    nodes,
+    InferenceError,
+)
+
+from codewatch.node_visitor import Inference
+
+
+def inference(node, fn, predicate=None):
+    return Inference(node, fn, predicate)
+
+
+DJANGO_MANAGER_METHODS_LIST = (
+    'all',
+    'annotate',
+    'bulk_create',
+    'distinct',
+    'exclude',
+    'filter',
+    'order_by',
+    'prefetch_related',
+    'select_related',
+    'using',
+)
+
+DJANGO_MANAGER_METHODS = (
+    'first',
+    'last',
+    'get',
+    'create',
+    'latest',
+    'get_or_create',
+    'update_or_create',
+) + DJANGO_MANAGER_METHODS_LIST
+
+
+CODEWATCH_MODEL_INFERENCE_KEY = '_codewatch_inferred_model_klass_obj'
+
+
+def get_inference_for_model(model_qname):
+    def _did_we_infer(node):
```

Comment:
Name is a little surprising given the tuple returned. Maybe:

```suggestion
     def _get_inference_details(node):
```

?